### PR TITLE
Added test stubs for Katello-certs-check utility

### PR DESCRIPTION
**Pull request by ntkathole**
Issue #5564

### CODE REVIEW

File: tests/foreman/sys/test_katello_certs_check.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+from robottelo.decorators import destructive, stubbed, tier1
+from robottelo.test import TestCase
+
+
+class KatelloCertsCheckTestCase(TestCase):
+    """Implements ``katello-certs-check`` tests"""
+
+    @stubbed()
+    @tier1
+    def test_positive_check_katello_certs_check_output(self):
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: `test_positive_validate_katello_certs_check_output`

File: tests/foreman/sys/test_katello_certs_check.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        """Check katello-certs-check generate correct command
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: Validate that katello-certs-check command generate correct output

File: tests/foreman/sys/test_katello_certs_check.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+        :steps:
+
+            1. Generate custom certs
+            2. Run katello-certs-check with the required valid arguments
+               katello-certs-check -c CERT_FILE -k KEY_FILE -r REQ_FILE
+               -b CA_BUNDLE_FILE
+            3. Assert the output has correct commands with options
+
+        :expectedresults: Katello-certs should generate correct commands with
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]:
```
:expectedresults: Katello-certs should generate correct commands with
    options.
```

File: tests/foreman/sys/test_katello_certs_check.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+                          options.
+
+        :caseautomation: notautomated
+        """
+
+    @destructive
+    @stubbed()
+    def test_positive_update_katello_certs(self):
+        """Update certificates on a currently running satellite installation
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: `Update certificates on a currently running satellite instance`

File: tests/foreman/sys/test_katello_certs_check.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+            4. Assert output is having correct capsule-certs-generate commands
+
+        :expectedresults: Katello-certs should be updated.
+
+        :caseautomation: notautomated
+        """
+
+    @destructive
+    @stubbed()
+    def test_positive_update_capsule_Katello_certs_using_absolute_path(self):
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: `test_positive_update_capsule_certs_using_absolute_path`
or
`test_positive_update_capsule_katello_certs_using_absolute_path`

File: tests/foreman/sys/test_katello_certs_check.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+               for --certs-tar
+
+        :expectedresults: Katello-certs should be updated.
+
+        :caseautomation: notautomated
+        """
+
+    @destructive
+    @stubbed()
+    def test_positive_update_capsule_Katello_certs_using_relative_path(self):
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: same here

File: tests/foreman/sys/test_katello_certs_check.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        """Check expiration of certificate
+
+        :id: 0acce44f-ebe5-42e1-a74b-3d4d20b97946
+
+        :steps:
+
+            1. Use expired certificate
+            2. Run katello-certs-check with the required arguments
+
+        :expectedresults: Checking expiration of certificate check should fail.
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: You should get an error that certificate is expired or what?

File: tests/foreman/sys/test_katello_certs_check.py [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+
+        :caseautomation: notautomated
+        """
+
+    @stubbed()
+    @tier1
+    def test_negative_check_ca_bundle(self):
+        """Check ca bundle file has correct data
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: if it is negative scenario then it is more `Check ca bundle file that contains invalid data`

File: tests/foreman/sys/test_katello_certs_check.py [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+        """Validate certificate subject
+
+        :id: 4df45b22-d077-470e-a786-2be329cd68a7
+
+        :steps:
+
+            1. Have a certificate with invalid subject
+            2. Run katello-certs-check with the required arguments
+
+        :expectedresults: Check for validating the certificate subject should
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]:
```
:expectedresults: Check for validating the certificate subject should
    fail.
```

File: tests/foreman/sys/test_katello_certs_check.py [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+        :expectedresults: Check for non ascii character should fail gracefully
+                          e.g. no traces.
+
+        :caseautomation: notautomated
+        """
+
+    @stubbed()
+    @tier1
+    def test_positive_check_katello_certs_check_without_req_file_output(self):
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: `test_positive_validate_without_req_file_output`?

File: tests/foreman/sys/test_katello_certs_check.py [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+
+        :steps:
+
+            1. Generate custom certs
+            2. Run katello-certs-check with the required valid arguments and
+               without -r REQ_FILE option
+               katello-certs-check -c CERT_FILE -k KEY_FILE -b CA_BUNDLE_FILE
+            3. Assert the output has correct commands with options
+
+        :expectedresults: Katello-certs should generate correct commands with
```
Comment (oshtaier) [inline tests/foreman/sys/test_katello_certs_check.py]: same here

File: tests/foreman/sys/test_katello_certs_check.py (hunk 7 above)
Comment (ntkathole) [reply tests/foreman/sys/test_katello_certs_check.py]: The check should fail. Expired certificate raises an error:
$ katello-certs-check -c server.invalid.crt -k server.key -r server.csr  -b ca/rootCA.pem
Checking expiration of certificate: [FAIL]
The certificate "/root/1419842/server.invalid.crt" already expired on: Nov 17 07:44:36 2016
Checking expiration of CA bundle: [OK]
Validating the certificate subject= /C=XX/L=Default City/O=Default Company Ltd/CN=example.com
Checking to see if the private key matches the certificate: [OK]
Checking ca bundle against the cert file: [OK]

**Comment by ntkathole** [comment]
@oshtaier comment addressed.Thanks

File: tests/foreman/sys/test_katello_certs_check.py [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+               for --certs-tar
+
+        :expectedresults: Katello-certs should be updated.
+
+        :caseautomation: notautomated
+        """
+
+    @destructive
+    @stubbed()
+    def test_positive_update_capsule_certs_using_relative_path(self):
```
Comment (san7ket) [inline tests/foreman/sys/test_katello_certs_check.py]: 👍

File: tests/foreman/sys/test_katello_certs_check.py [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+
+        :expectedresults: Katello-certs-check should generate correct commands
+         with options.
+
+        :caseautomation: notautomated
+        """
+
+    @destructive
+    @stubbed()
+    def test_positive_update_katello_certs(self):
```
Comment (san7ket) [inline tests/foreman/sys/test_katello_certs_check.py]: @upgrade

File: tests/foreman/sys/test_katello_certs_check.py (hunk 12 above)
Comment (san7ket) [reply tests/foreman/sys/test_katello_certs_check.py]: upgrade

**Comment by ntkathole** [comment]
@oshtaier ACK please.

**Pull request merged**